libc/calls/commandv.c

```diff
@@ -57,7 +64,16 @@ static bool AccessCommand(const char *name, char *path, size_t pathsz,
   }
   memcpy(path + pathlen, name, namelen);
   memcpy(path + pathlen + namelen, suffix, suffixlen + 1);
-  if (!access(path, X_OK)) return true;
+  if (!access(path, X_OK)) {
+    struct stat st;
+    if (!stat(path, &st)) {
+      if (S_ISREG(st.st_mode)) {
+        return true;
+      } else {
+        errno = EISDIR;
```

Comment:
Nitpick: @jart did give `EISDIR` as an example, but `EACCES` looks like a better fit to me as `execve` uses `EACCES` for "The file or a script interpreter is not a regular file." and just because it isn't a regular file doesn't mean it's a directory. https://man7.org/linux/man-pages/man2/execve.2.html#ERRORS

Reply:
It isn't obvious to me what causes the difference in behavior, I can take a look at it either this evening or tomorrow. `EACCES` could be useful if they thought a executable with that name was in path, but there is only a directory, `ENOENT` seems wrong if anything exists by that name.

Reply:
I don't think it's a showstopper, but I marked it for me to look into it #762